redwood-treehouse/src/commonTest/kotlin/app/cash/redwood/treehouse/StateSnapshotTest.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.treehouse
+
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import kotlin.test.Test
+import kotlin.test.assertTrue
+import kotlinx.serialization.json.JsonPrimitive
+
+class StateSnapshotTest {
+
+  @Test
+  fun toValueMapWorksAsExpected() {
+    val stateSnapshot = stateSnapshot()
+    val valuesMap = stateSnapshot.toValuesMap()
+    assertThat(valuesMap.entries.size).isEqualTo(4)
+    assertTrue(valuesMap["key1"]!![0] is MutableState<*>)
+    assertThat((valuesMap["key1"]!![0] as MutableState<*>).value).isEqualTo(JsonPrimitive(1))
+
+    assertThat(valuesMap["key2"]).isEqualTo(listOf(1.0))
+
+    assertThat(valuesMap["key3"]!![0] is MutableState<*>)
+    assertThat((valuesMap["key3"]!![0] as MutableState<*>).value).isEqualTo(JsonPrimitive("str"))
+
+    assertThat(valuesMap["key4"]).isEqualTo(listOf("str"))
```

Comment:
Could these test cases be simplified with the [`containsOnly` function](https://willowtreeapps.github.io/assertk/assertk/assertk.assertions/contains-only.html)?

```kotlin
    assertThat(valuesMap).containsOnly(
        "key1" to listOf(mutableStateOf(JsonPrimitive(1)),
        …
    )
```

Reply:
`containsOnly` errors out: 

`expected to contain only:<{"key1"=[MutableState(value=1)@916835004], "key2"=[1.0], "key3"=[MutableState(value="str")@2108297149], "key4"=["str"]}> but was:<{"key1"=[MutableState(value=1)@1112737073], "key2"=[1.0], "key3"=[MutableState(value="str")@1513867245], "key4"=["str"]}>
 elements not found:<{"key1"=[MutableState(value=1)@916835004], "key3"=[MutableState(value="str")@2108297149]}>
 extra elements found:<{"key1"=[MutableState(value=1)@1112737073], "key3"=[MutableState(value="str")@1513867245]}>`

for the `MutableState` ones, `containsOnly` doesn't unwrap and compare the values (it compares the instances)